src/errors/runtime/index.js

```diff
@@ -115,3 +115,20 @@ export class CompositeError extends Error {
         this.code  = RUNTIME_ERRORS.compositeArgumentsError;
     }
 }
+
+export class ReporterPluginError extends Error {
+    constructor (originalError) {
+        const template     = TEMPLATES[RUNTIME_ERRORS.uncaughtErrorInReporter];
+        const errorMessage = originalError.toString();
+
+        super(renderTemplate(template, errorMessage));
```

Comment:
I remember we had a conversation about subclassing from the `GeneralError`, since it renders the template.